bc_obps/registration/api/utils/operator_utils.py

```diff
@@ -0,0 +1,106 @@
+from django.db import transaction
+from registration.utils import update_model_instance
+from typing import Optional
+from registration.models import (
+    User,
+    Operator,
+    User,
+    UserOperator,
+)
+from registration.schema.user_operator import UserOperatorOperatorIn
+from registration.models import (
+    Operator,
+    User,
+    UserOperator,
+    Address,
+)
+
+
+# ADDRESS HELPERS
+class AddressesData:
+    physical_street_address: str
+    physical_municipality: str
+    physical_province: str
+    physical_postal_code: str
+    mailing_address_same_as_physical: bool
+    mailing_street_address: Optional[str]
+    mailing_municipality: Optional[str]
+    mailing_province: Optional[str]
+    mailing_postal_code: Optional[str]
+
+
+def handle_operator_addresses(address_data: AddressesData, physical_address_id, mailing_address_id, prefix=""):
+    # create or update physical address record
+    physical_address, _ = Address.objects.update_or_create(
+        id=physical_address_id,
+        defaults={
+            "street_address": address_data.get(f'{prefix}physical_street_address'),
+            "municipality": address_data.get(f'{prefix}physical_municipality'),
+            "province": address_data.get(f'{prefix}physical_province'),
+            "postal_code": address_data.get(f'{prefix}physical_postal_code'),
+        },
+    )
+    if address_data.get(f'{prefix}mailing_address_same_as_physical'):
+        mailing_address = physical_address
```

Comment:
There is still a possibility of orphaned addresses where they used to have both a physical and mailing address, but then change mailing_address_same_as_physical to True.

In this case, the id of the mailing address will be set to the id of the physical address, but the original mailing address record will remain orphaned in the database.

Should we add a condition here opposite to the condirion below where if the current mailing address id != physical address id then we prune the old mailing address? Should we archive or hard delete this mailing address?

Reply:
https://github.com/bcgov/cas-registration/issues/970